examples/js/exporters/GLTFExporter.js

```diff
@@ -964,6 +964,36 @@ THREE.GLTFExporter.prototype = {
 
 			}
 
+			if ( material.isGLTFSpecularGlossinessMaterial ) {
+
+				var specularExtensionName = "KHR_materials_pbrSpecularGlossiness";
+				var specularMapDef = material.specularMap ? { index: processTexture( material.specularMap ) } : undefined;
+				var diffuseMapDef = material.map ? { index: processTexture( material.map ) } : undefined;
+				if ( material.specularMap ) {
+					
+					applyTextureTransform( specularMapDef, material.specularMap );
+			
+				}
+			
+				// alpha default is 1, rgb will be overridden
+				var diffuseFactor = [ 1, 1, 1, 1 ];
+				material.color.toArray( diffuseFactor, 0 );
+				// same story for specularFactor
+				var specularFactor = [ 1, 1, 1 ];
+				material.specular.toArray( specularFactor, 0 );
+			
+				gltfMaterial.extensions = {};
+				gltfMaterial.extensions[ specularExtensionName ] = {
+					"diffuseFactor": diffuseFactor,
+					"diffuseTexture": diffuseMapDef,
+					"specularFactor": specularFactor,
+					"glossinessFactor": material.glossiness,
+					"specularGlossinessTexture": specularMapDef
+				};
```

Comment:
Could you reorganize this to match the rest of the `processMaterial` method, and reuse values where possible? I think it's easier to read when the method is consistent throughout, and there are important differences in e.g. opacity.

```js
var specularExtension = {};

if ( gltfMaterial.pbrMetallicRoughness.baseColorFactor ) {

    specularExtension.diffuseFactor = gltfMaterial.pbrMetallicRoughness.baseColorFactor;

}

if ( gltfMaterial.pbrMetallicRoughness.baseColorTexture ) {

    specularExtension.diffuseTexture = gltfMaterial.pbrMetallicRoughness.baseColorTexture;

}
```

Reply:
I've reorganized the code in the last commit so it is easier to read through the processMaterial method.